bolt-sidecar/src/state/execution.rs

```diff
@@ -210,6 +214,24 @@ impl<C: StateFetcher> ExecutionState<C> {
             }
         }
 
+        if let Some(template) = self.get_block_template(target_slot) {
+            // Check if the committed gas exceeds the maximum
+            if template.committed_gas() + req.tx.gas_limit()
+                >= self.max_committed_gas_per_slot.get()
+            {
+                return Err(ValidationError::MaxCommittedGasReachedForSlot(
+                    self.slot,
+                    self.max_committed_gas_per_slot.get(),
+                ));
+            }
+        } else if req.tx.gas_limit() >= self.max_committed_gas_per_slot.get() {
+            // Check if the committed gas limit itself exceeds the maximum when template is None
+            return Err(ValidationError::MaxCommittedGasReachedForSlot(
+                self.slot,
+                self.max_committed_gas_per_slot.get(),
+            ));
+        }
+
```

Comment:
This check is for the case when no template is built and the request itself exceeds the limit

Reply:
This can be refactored a bit along this lines!
```suggestion
        // Check if the committed gas exceeds the maximum
        let template_committed_gas = self.get_block_template(target_slot).map(|t| t.committed_gas()).unwrap_or(0);
        if template_committed_gas + req.tx.gas_limit() >= self.max_committed_gas_per_slot.get() {
            return Err(ValidationError::MaxCommittedGasReachedForSlot(
                self.slot,
                self.max_committed_gas_per_slot.get(),
            ));
        }
```